modules/dataframe/src/properties/colormapproperty.cpp

```diff
@@ -0,0 +1,182 @@
+    , nColors(rhs.nColors) {
+
+    updateColormaps();
+    type.onChange([&]() { updateColormaps(); });
+    diverging.onChange([&]() { updateColormaps(); });
+
+    colormap.onChange([&]() {
+        nColors.set(*nColors, getMinNumberOfColorsForFamily(colormap),
+                    getMaxNumberOfColorsForFamily(colormap), 1);
+    });
+
+    diverging.visibilityDependsOn(
+        type, [](auto prop) -> bool { return prop == ColormapType::Continous; });
+    divergenceMidPoint.visibilityDependsOn(diverging, [](auto prop) -> bool { return prop; });
+    // Always discrete for categorical data
+    discrete.visibilityDependsOn(type, [](auto prop) { return prop == ColormapType::Continous; });
+
+    addProperty(type);
+    addProperty(colormap);
+    addProperty(diverging);
+    addProperty(divergenceMidPoint);
+
+    addProperty(discrete);
+    addProperty(nColors);
+}
+
+ColormapProperty* ColormapProperty::clone() const { return new ColormapProperty(*this); }
+
+colorbrewer::Category ColormapProperty::getCategory() const {
+    colorbrewer::Category cat;
+    switch (type) {
+        case ColormapType::Continous:
+            cat = diverging ? colorbrewer::Category::Diverging : colorbrewer::Category::Sequential;
+            break;
+        case ColormapType::Categorical:
+            cat = colorbrewer::Category::Qualitative;
+            break;
+    }
+    return cat;
+}
+
+colorbrewer::Family ColormapProperty::getFamily() const { return *colormap; }
+
+void ColormapProperty::setupForColumn(const Column& col) {
+    col.getBuffer()->getRepresentation<BufferRAM>()->dispatch<void, dispatching::filter::Scalars>(
+        [&](auto ram) -> void {
+            using T = typename util::PrecisionValueType<decltype(ram)>;
+            auto minMax = util::bufferMinMax(ram, IgnoreSpecialValues::Yes);
+            setupForColumn(col, minMax.first.x, minMax.second.x);
+        });
+}
+void ColormapProperty::setupForColumn(const Column& col, double minVal, double maxVal) {
+    NetworkLock lock(this);
+
+    if (auto catCol = dynamic_cast<const CategoricalColumn*>(&col)) {
+        type.set(ColormapType::Categorical);
+        colormap.set(colorbrewer::Family::Paired);
+        auto maxColors = getMaxNumberOfColorsForFamily(colormap);
+        auto numCategories = catCol->getCategories().size();
+        if (maxColors < numCategories) {
+            LogWarn("Categories exceed maximum classes in colormap. "
+                    << maxColors << " will be used but " << numCategories
+                    << " needed. Override colormap to provide your own colormap with more classes.")
+        }
+        nColors = numCategories;
+        discrete = true;
+    } else {
+        type.set(ColormapType::Continous);
+        // Better contrast when using many classes
+        nColors = getMaxNumberOfColorsForFamily(colormap);
+        discrete = false;
+    }
+    divergenceMidPoint.set(0.5 * (minVal + maxVal), minVal, maxVal, 0.1 * (maxVal - minVal));
```

Comment:
is 10% really a good step value? Shouldn't it be rather 0.01?

Reply:
Would it make sense to set it to divergenceMidPoint zero instead of `(0.5 * (minVal + maxVal)` when `minVal < 0` and  `maxVal > 1`?

Reply:
That might be a good initial guess in this case!

Reply:
> is 10% really a good step value? Shouldn't it be rather 0.01?

As good as any ;) Changed it to 0.01

> Would it make sense to set it to divergenceMidPoint zero instead of (0.5 * (minVal + maxVal) when minVal < 0 and maxVal > 1?

Good point. Let's try it and see how it works